include/deal.II/grid/grid_tools.h

```diff
@@ -756,6 +756,86 @@ namespace GridTools
   std::vector<typename MeshType::active_cell_iterator>
   compute_ghost_cell_halo_layer (const MeshType &mesh);
 
+  /**
+   * Extract and return the active cell within a @p layer_thickness
+   * around a subdomain (set of active cells) in the @p mesh.
+   * Here, the "subdomain" consists of exactly all of
+   * those cells for which the @p predicate returns @p true.
+   *
+   * The function first computes interior active, boundary cells of the
+   * subdomain that conform to the predicate.
```

Comment:
Do you mean something like "the cells that form the 'surface' of the subdomain that consists of all of the active cells for which the predicate is true"?